java/arcs/core/tools/Proto2Schema.kt

```diff
@@ -0,0 +1,138 @@
+package arcs.core.tools
+
+import arcs.core.data.FieldType
+import arcs.core.data.RecipeEnvelopeProto
+import arcs.core.data.Schema
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.arguments.argument
+import com.github.ajalt.clikt.parameters.arguments.multiple
+import com.github.ajalt.clikt.parameters.options.default
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.types.file
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.CodeBlock
+import com.squareup.kotlinpoet.PropertySpec
+import java.io.File
+
+class Proto2Schema : CliktCommand(
+    help = """Generates Schemas and Types from a protobuf-serialized manifest.
+    
+    This script reads schemas from a serialized manifest and generates Kotlin `Schema` and `Type` 
+    classes.""",
+    printHelpOnEmptyArgs = true
+) {
+
+    val outdir by option(help = "output directory; defaults to '.'").file(fileOkay = false)
+    val outfile by option(help = "output filename; if omitted")
+    val packageName by option(help = "scope to specified package; default: 'arcs'").default("arcs")
+    val protos by argument(help = "paths to protobuf-serialized manifests")
+        .file(exists = true).multiple()
+
+    /** Mock Proto2Schema: So far, this is equivalent to copy. */
+    override fun run() {
+        protos.forEach { protoFile ->
+            val outFile = outputFile(protoFile)
+            echo("$protoFile --> $outFile")
+
+            val bytes = protoFile.readBytes()
+            val envelope = RecipeEnvelopeProto.parseFrom(bytes)
+
+            outFile.writeBytes(protoFile.readBytes())
+        }
+    }
+
+    /** Produces a File object per user specification, or with default values. */
+    fun outputFile(inputFile: File): File {
+        val outputName = outfile ?: inputFile.nameWithoutExtension + ".kt"
+        val outputPath = outdir ?: System.getProperty("user.dir")
+        return File("$outputPath/$outputName")
+    }
+
+    fun formatName(name: String): String {
+        return name[0].toLowerCase() + name.substring(1)
+    }
+
+    fun generateSchemas(schemas: List<Schema>): Iterable<PropertySpec> {
+        var anons = 0
+        val schemaClass = ClassName("arcs.core.data", "Schema")
+        val schemaNameClass = ClassName("arcs.core.data", "SchemaName")
+        val schemaFieldsClass = ClassName("arcs.core.data", "SchemaFields")
```

Comment:
If this tools package can depend on `//java/arcs/core/data`, prefer using Kotlin reflection (will need a new maven library import in the `WORKSPACE`/third_party for this) and KotlinPoet's `.asTypeName`:

```kotlin
val schemaClass = Schema::class.createType().asTypeName()
```

This will allow us to more easily and safely refactor the classes you're using here.

Reply:
I think you could probably make that shorter using an extension function like Schema::class.toTypeName()